components/automate-ui/src/app/pages/policy/add-members/policy-add-members.spec.ts

```diff
@@ -0,0 +1,112 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { ReactiveFormsModule, FormBuilder, Validators } from '@angular/forms';
+import { StoreModule } from '@ngrx/store';
+import { MockComponent } from 'ng2-mock-component';
+
+import { Regex } from 'app/helpers/auth/regex';
+import { ChefPipesModule } from 'app/pipes/chef-pipes.module';
+import { customMatchers } from 'app/testing/custom-matchers';
+import { policyEntityReducer } from 'app/entities/policies/policy.reducer';
+import { PolicyAddMembersComponent } from './policy-add-members.component';
+
+
+
+
+
+describe('PolicyAddMembersComponent', () => {
+    let component: PolicyAddMembersComponent;
+    let fixture: ComponentFixture<PolicyAddMembersComponent>;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            declarations: [
+                MockComponent({ selector: 'chef-page' }),
+                MockComponent({ selector: 'chef-modal', inputs: ['visible'] }),
+                MockComponent({ selector: 'chef-form-field' }),
+                MockComponent({ selector: 'chef-error'}),
+                MockComponent({ selector: 'chef-button', inputs: ['disabled'] }),
+                MockComponent({ selector: 'chef-input'}),
+                MockComponent({ selector: 'chef-table' }),
+                MockComponent({ selector: 'chef-thead' }),
+                MockComponent({ selector: 'chef-tr' }),
+                MockComponent({ selector: 'chef-th' }),
+                MockComponent({ selector: 'chef-tbody' }),
+                MockComponent({ selector: 'chef-td' }),
+                MockComponent({ selector: 'chef-checkbox', inputs: ['checked'] }),
+                MockComponent({ selector: 'a', inputs: ['routerLink'] }),
+                PolicyAddMembersComponent
+            ],
+            imports: [
+                ChefPipesModule,
+                ReactiveFormsModule,
+                RouterTestingModule,
+                StoreModule.forRoot({
+                    policies: policyEntityReducer
+                })
+            ]
+        }).compileComponents();
+    }));
+
+    beforeEach(() => {
+        jasmine.addMatchers(customMatchers);
+        fixture = TestBed.createComponent(PolicyAddMembersComponent);
+        component = fixture.componentInstance;
+        component.expressionForm = new FormBuilder().group({
+            type: ['', [Validators.required, Validators.pattern(Regex.patterns.NON_BLANK)]],
+            identity: [''],
+            name: ['']
+        });
+        fixture.detectChanges();
+    });
+
+    it('should be created', () => {
+        expect(component).toBeTruthy();
+    });
+
```

Comment:
Nice tests; let me suggest to see if you could combine most of these into one or perhaps two sets of data-driven tests. Search the code base for `using([` and you will find many examples.

Reply:
Thanks for the suggestion!  `using` is awesome! 🎉